Fix SINET

**Describe the bug**
SINET model doesn't work properly.

**To Reproduce**
Just use the SINET model.

**Expected behavior**

A reasonable segmentation.

**Screenshots**

I can't make a screenshot since I'm on my phone. But it wasn't working for me yesterday.

**Log and Crash Report**
It doesn't crash. Just the segmentation is off.


**Desktop (please complete the following information):**

 - OS: macos
 - Plugin Version: 1.0.0
 - OBS Version: 29.1

**Additional context**

Add any other context about the problem here.

I got the following error

```
error: [obs-backgroundremoval] OpenCV(4.7.0) /Users/runner/work/obs-backgroundremoval/obs-backgroundremoval/plugin/build_arm64/OpenCV_Build-prefix/src/OpenCV_Build/modules/imgproc/src/contours.cpp:197: error: (-210:Unsupported format or combination of formats) [Start]FindContours supports only CV_8UC1 images when mode != CV_RETR_FLOODFILL otherwise supports CV_32SC1 images only in function 'cvStartFindContours_Impl'
```

--- a/src/models/ModelSINET.h
+++ b/src/models/ModelSINET.h
@@ -10,10 +10,26 @@ class ModelSINET : public ModelBCHW {
 
   virtual void prepareInputToNetwork(cv::Mat &resizedImage, cv::Mat &preprocessedImage)
   {
-    cv::subtract(resizedImage, cv::Scalar(102.890434, 111.25247, 126.91212), resizedImage);
-    cv::multiply(resizedImage, cv::Scalar(1.0 / 62.93292, 1.0 / 62.82138, 1.0 / 66.355705) / 255.0,
-                 resizedImage);
+    resizedImage = (resizedImage - cv::Scalar(102.890434, 111.25247, 126.91212)) /
+      cv::Scalar(62.93292 * 255.0, 62.82138 * 255.0, 66.355705 * 255.0);
     hwc_to_chw(resizedImage, preprocessedImage);
+  }
+
+  virtual cv::Mat getNetworkOutput(const std::vector<std::vector<int64_t>> &outputDims,
+                                   std::vector<std::vector<float>> &outputTensorValues)
+  {
+    UNUSED_PARAMETER(outputDims);
+    return cv::Mat(320, 320, CV_32FC2, outputTensorValues[0].data());
+  }
+
+  virtual void postprocessOutput(cv::Mat &outputImage)
+  {
+    cv::Mat outputTransposed;
+    chw_to_hwc_32f(outputImage, outputTransposed);
+    // take 2nd channel
+    std::vector<cv::Mat> outputImageSplit;
+    cv::split(outputTransposed, outputImageSplit);
+    outputImage = outputImageSplit[1];
   }
 };
 

--- a/src/models/ModelSINET.h
+++ b/src/models/ModelSINET.h
@@ -11,7 +11,7 @@ class ModelSINET : public ModelBCHW {
   virtual void prepareInputToNetwork(cv::Mat &resizedImage, cv::Mat &preprocessedImage)
   {
     resizedImage = (resizedImage - cv::Scalar(102.890434, 111.25247, 126.91212)) /
-      cv::Scalar(62.93292 * 255.0, 62.82138 * 255.0, 66.355705 * 255.0);
+                   cv::Scalar(62.93292 * 255.0, 62.82138 * 255.0, 66.355705 * 255.0);
     hwc_to_chw(resizedImage, preprocessedImage);
   }
 

